app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityPresenter.kt

```diff
@@ -92,56 +92,54 @@ class QuestionPlayerActivityPresenter @Inject constructor(
     val skillIds =
       activity.intent.getStringArrayListExtra(QUESTION_PLAYER_ACTIVITY_SKILL_ID_LIST_ARGUMENT_KEY)
     val startDataProvider =
-      questionTrainingController.startQuestionTrainingSession(profileId, skillIds)
-    startDataProvider.toLiveData().observe(
-      activity,
-      {
-        when {
-          it.isPending() -> oppiaLogger.d(
+      skillIds?.let { questionTrainingController.startQuestionTrainingSession(profileId, it) }
+    startDataProvider?.toLiveData()?.observe(
+      activity
+    ) {
+      when {
+        it.isPending() -> oppiaLogger.d(
+          "QuestionPlayerActivity",
+          "Starting training session"
+        )
+        it.isFailure() -> {
+          oppiaLogger.e(
```

Comment:
Ditto for this and other similar changes.

Reply:
okk